synfig-studio/src/gui/canvasview.cpp

```diff
@@ -2952,11 +2971,12 @@ CanvasView::on_waypoint_clicked_canvasview(ValueDesc value_desc,
 	case 2:
 	{
 		Gtk::Menu* waypoint_menu(manage(new Gtk::Menu()));
-		waypoint_menu->signal_hide().connect(sigc::bind(sigc::ptr_fun(&delete_widget), waypoint_menu));
+		menu_present= true;
```

Comment:
Why does it exist?

Reply:
menu_present (name should probably be more descriptive) is used as a flag to know if there is already a popup menu (the waypoint right click menu). and if it is present then the other popup menu (right click without waypoint) shouldnt also show.

Reply:
Got it.
Well, it wouldn't be necessary if you follow my suggestion that it's better you create a new signal for `Widget_TimeTrack` (or for `SelectDragHelper`)

Reply:
removed, no longer needed in new commit.